Microsoft/pxt
# Blockly April release

Pulling in the Blockly mainline April release.

To test: 

https://makecode.adafruit.com/app/631c8197d9aea7766e208b6b278380d45543fd0e-9b9afbd4cf

Testing
Test the following on various browsers:

###Zoom in / out
Use the mouse to scroll in / out.
Ctrl / Cmd + Scroll to zoom in / out.

###Toolbox
Able to select different categories and drag various blocks into the workspace.
Select a category that has a “More…” subcategory
Select the parent category and switch between them back and forth
Select the parent category, then select it again to hide it.
Select any category, and select it again to hide it.
Select the Advanced category to toggle advanced categories, and select it again to hide them.

###JS Card
Hover over any card, from the toolbox and the workspace. A help card should appear with the JS output.

###Typing a string
Select a block that inputs a string, and type in the string.

###Resize workspace
Resize the window, and make sure Blockly auto-sizes to fit the window.
Check what that looks like in a mobile view.

###Context Menu
Right click anywhere in the workspace, make sure our custom context menu appears with options to Delete, Shuffle and Download a screenshot.
Check that downloading a screenshot works.
Check that deleting / shuffling works.

###Javascript editor
Make sure you can switch to the Javascript editor and back and forth between the two editors.

###Sandbox Mode
Check sandbox mode for both the blocks editor and the Javascript editor.

###Custom Variables Flyout
Check that the custom variables flyout looks right, and that you’re able to create, delete, and change variables from within Blockly.

###Touch
